fuel_test/deploy_env.sh

```diff
@@ -0,0 +1,287 @@
+	local fm_ip="$1"
+	local env_name="$2"
+	local node_mac="$3"
+	local role="$4"
+	local interface_yaml="$5"
+
+	scp configure_interfaces.py root@$fm_ip:/tmp/
+	scp $interface_yaml root@$fm_ip:/tmp/
+	ssh -qo stricthostkeychecking=no root@$fm_ip \
+	'
+	export fuelclient_custom_settings="/etc/fuel/client/config.yaml"
+	env_id=$(fuel env | grep "'$env_name'" | egrep -o "^[0-9]+")
+	node_id=$(fuel node --node-id "'$node_mac'" | grep "'$node_mac'" | egrep -o "^[0-9]+")
+	fuel node set --node $node_id --env $env_id --role "'$role'"
+
+	cd /tmp
+	fuel node --node-id $node_id --net -d
+	./configure_interfaces.py node_$node_id/interfaces.yaml '$interface_yaml'
+	fuel node --node-id $node_id --net -u
+	'
+}
+
+function verify_network {
+	# Do network verification and wait for result
+	# return 1 when passes, return 0 when failed or retrial timeout
+	local fm_ip="$1"
+	local env_name="$2"
+	local retry_count=$3
+	local retry_interval=$4
+
+	ssh -qo stricthostkeychecking=no root@$fm_ip \
+	'
+	export fuelclient_custom_settings="/etc/fuel/client/config.yaml"
+	env_id=$(fuel env | grep "'$env_name'" | egrep -o "^[0-9]+")
+	fuel network --verify --env $env_id &> /dev/null
+	'
+
+	local counter=0
+	local ok
+	while [ $counter -lt $retry_count ]; do
+		task=$(ssh -qo StrictHostKeyChecking=no root@$fm_ip \
+		'
+		export FUELCLIENT_CUSTOM_SETTINGS="/etc/fuel/client/config.yaml"
+		fuel task | grep verify_networks
+		')
+		status=$(echo $task | awk -F '|' '{print $2}')
+		progress=$(echo $task | awk -F '|' '{print $5}')
+		if [ $status == "error" ]; then
+			echo 0
+			return
+		fi
+		if [ $progress -eq "100" ]; then
+			echo 1
+			return
+		fi
+		let counter=counter+1
+		sleep $retry_interval
+	done
+	echo 0
+}
+
+function deploy_env {
+	local fm_ip="$1"
+	local env_name="$2"
+
+	ssh -qo stricthostkeychecking=no root@$fm_ip \
+	'
+	export fuelclient_custom_settings="/etc/fuel/client/config.yaml"
+	env_id=$(fuel env | grep "'$env_name'" | egrep -o "^[0-9]+")
+	fuel deploy-changes --env $env_id
+	'
+}
+
+FM_IP=$(wait_for_fm "$XS_HOST" "$FM_NAME" 10 60)
+[ -z $FM_IP ] && echo "Fuel Master IP obtaining timeout" && exit -1
+
+build_plugin $FM_IP $FUEL_PLUGIN_REFSPEC
+echo "Fuel plugin with $FUEL_PLUGIN_REFSPEC is built"
+
+NAILGUN_READY=$(wait_for_nailgun "$FM_IP" 10 60)
+[ $NAILGUN_READY -eq 0 ] && echo "Nailgun test connection timeout" && exit -1
+
+install_plugin "$FM_IP"
+echo "Fuel plugin is installed"
+
+create_env "$FM_IP" "$ENV_NAME" "$REL_NAME" "$ATTRIBUTES_YAML" "$NETWORK_YAML"
+
+start_node "$XS_HOST" "Compute"
+echo "Compute Node is started"
+start_node "$XS_HOST" "Controller"
+echo "Controller Node is started"
+
+COMPUTE_MAC=$(get_node_mac "$XS_HOST" "Compute")
+[ -z $COMPUTE_MAC ] && echo "Compute node doesnot exist" && exit -1
+COMPUTE_DISCOVERED=$(wait_for_node "$FM_IP" "$COMPUTE_MAC" 10 60)
+[ -z $COMPUTE_DISCOVERED ] && echo "Compute node discovery timeout" && exit -1
+add_env_node "$FM_IP" "$ENV_NAME" "$COMPUTE_MAC" "compute,cinder" $INTERFACE_YAML
```

Comment:
I think the main three scripts that uses the localrc variables should source it directly rather than assuming it was sourced from fuel_test.sh - this would allow us to invoke each stage separately if needed.